flink-runtime/src/test/java/org/apache/flink/runtime/io/network/buffer/BufferBuilderAndConsumerTest.java

```diff
@@ -81,6 +81,24 @@ public void multipleAppends() {
 		assertContent(bufferConsumer, 0, 1, 2, 3, 42);
 	}
 
+	@Test
+	public void multipleFinish() {
+		BufferBuilder bufferBuilder = createBufferBuilder();
+		BufferConsumer bufferConsumer = bufferBuilder.createBufferConsumer();
+
+		bufferBuilder.appendAndCommit(toByteBuffer(42, 44));
+
+		assertFalse(bufferBuilder.isFinished());
+
+		bufferBuilder.finish();
+		assertTrue(bufferBuilder.isFinished());
+		bufferBuilder.finish();
+		assertTrue(bufferBuilder.isFinished());
+
+		assertContent(bufferConsumer, 42, 44);
```

Comment:
Can you also add a content check like this into `testIsFinished()` because there is no direct test that a finished buffer and/or consumer actually has the contents it is supposed to have.